### ISSUE TITLE
Error: Can not get job status

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#568]
In GitLab by @molidg on Aug 7, 2020, 10:01

Hi,
I'm using autosubmit from the bscesoper machine to run the operationals and the last days I'm having a problem with the run for the ICAP (a0yh) and the BDFC (a2in). Autosubmit is not able to get the status of the jobs. The error is this:

```
[WARNING] Retrying check job command: sacct -n -j 11199473 -o "State"
[ERROR] Can not get job status for job id (11199473), retrying in 10 sec
```
The problems is that the 'create' command is failing the next day because autosubmit is not available (is running the same expID).

Could you help me? Thanks!

FYI: @mcastril @kserradell @gmontanepinto

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 7, 2020, 10:36

Hello

Apparently, the command can be send and iut successfully completes but AS doesn't recognize it as is worked properly not sure how. 

```
[WARNING] 2020-08-07 08:07:33,873 Retrying check job command: sacct -n -j 11199473 -o "State"
[ERROR] 2020-08-07 08:07:33,873 Can not get job status for job id (11199473), retrying in 10 sec
2020-08-07 08:07:50,812 Command sacct -n -j 11199473 -o "State" in mn4 successful with out message: 
[WARNING] 2020-08-07 08:07:50,812 Retrying check job command: sacct -n -j 11199473 -o "State"
[ERROR] 2020-08-07 08:07:50,812 Can not get job status for job id (11199473), retrying in 10 sec
2020-08-07 08:08:07,756 Command sacct -n -j 11199473 -o "State" in mn4 successful with out message:    PENDING 
   RUNNING 

[WARNING] 2020-08-07 08:08:07,756 Retrying check job command: sacct -n -j 11199473 -o "State"
[ERROR] 2020-08-07 08:08:07,756 Can not get job status for job id (11199473), retrying in 10 sec
2020-08-07 08:08:24,749 Command sacct -n -j 11199473 -o "State" in mn4 successful with out message:  COMPLETED 
 COMPLETED 
```

This is the log previous to this message, as you can see:

This line  `020-08-07 08:08:07,756 Can not get job status for job id (11199473), retrying in 10 sec`

Implies that the command didn't work however this is false given that it also indicates that
```
2020-08-07 08:08:07,756 Command sacct -n -j 11199473 -o "State" in mn4 successful with out message:    PENDING 
   RUNNING 
```
The command was sucessful.

The last log, shown by @molidg, can be misleading since this is the output of when the job finishes and there is no more output to be retrieved, but in the previous step, this was retrieving output. Also, it should retry the job 5 times and seems that it goes forever.

So basically, it is working but is not being acknowledged not sure how.

So I need additional info like:

this is being launched on the operational machine, right? I don't have experience with this machine so:

Which version of autosubmit are you using? Seems that is 3.12.0 or lower, this AS version is from the modules like normal machines,  or was previously installed and is not being updated? 

In what machine is being launched? has any difference with the autosubmit virtual machines? (terminal,distribution)  had it any changes? 

 How long has been this happening? only for a week? @molidg  


I did a copy of the experiment a0yh with dummy templates ( None option) and it worked flawlessly on bscearth375. (the experiment is a31d)

**Comment by kinow** [comment]
In GitLab by @molidg on Aug 7, 2020, 10:49

Hi @dbeltrankyl , the version of autosubmit is 3.12.0 but I'm not sure if this version is the same as the version installed in the other machines. @kserradell could you help @dbeltrankyl with the information about the virtual machines? 
And finally I had this problem with old versions of autosubmit but the error didn't dissapear when we updated the version of autosubmit.

**Comment by kinow** [comment]
In GitLab by @kserradell on Aug 7, 2020, 10:53

bsesoper is a virtual machine running an SLES OS but it has the same software stack than WS. Is accessible to all users.
```
[kserrade@ bscearth408 ~ ]$ ssh bscesoper.bsc.es
Last login: Fri Aug  7 10:50:19 2020 from 84.88.185.148

+----------------------------------------------------------+
   .-.--_
 ,´,´.´   `.
 | | | BSC |     BSC-CNS Barcelona Supercomputing Center
 `.`.`. _ .´
   `·`··
+----------------------------------------------------------+
Welcome to bscesoper. - 84.88.185.105
* Distro: SLES 12.3
* Kernel: 4.4.138-94.39-default
* Synced: 2020-07-09T09:30:55Z
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 7, 2020, 12:39

Hi @molidg 

This happens always? I'm trying to replicate your issues but I'm not able to

**Comment by kinow** [comment]
In GitLab by @molidg on Aug 7, 2020, 12:40

no... now I'm running the a0yh experiment with the same script as usual and it's working U_U

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 7, 2020, 16:26

Firstly I though about a kind of connection issue between bscesoper.bsc.es and MN4 (firewalled?). But this:

```
[WARNING] 2020-08-07 08:54:35,584 Retrying check job command: sacct -n -j 11199473 -o "State"
[ERROR] 2020-08-07 08:54:35,584 Can not get job status for job id (11199473), retrying in 10 sec
2020-08-07 08:54:52,673 Command sacct -n -j 11199473 -o "State" in mn4 successful with out message:  COMPLETED
```

Indicates that it indeed can retrieve the output, as you pointed out @dbeltrankyl . If AS is showing the output (COMPLETED), why it does not recognize it?

Could it be any special character that is not being seen in the prints but it's different from the string "COMPLETED"? But why then it would happen only sometimes?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 13, 2020, 13:13

Hello

I still didn't detect the issue, however, if autosubmit detect a hang ( by forcing a timeout with the help of the log and exceptions refactor), it will be automatically restarted and hopefully, it will continue correctly.

I'll update the issue when it is uploaded into **3.13.0b** (more info at #570) 

Note: I know that you are currently using 3.12.0, sadly, for that version the issue will remain since the changes are too large to be implemented in that version.

**Comment by kinow** [comment]
In GitLab by @molidg on Aug 13, 2020, 13:34

Hi @dbeltrankyl, ok. at the moment I will kill autosubmit before create the experiment again. W 
 ThankS!
I can close the issue

**Comment by kinow** [comment]
In GitLab by @ mtpay on Sep 18, 2020, 08:45

I was about opening this issue and I saw that @molidg reported the same problem in the nohup:
```
[WARNING] Retrying check job command: sacct -n -j 11679249 -o "State"
[ERROR] Can not get job status for job id (11679249), retrying in 10 sec
[ERROR] Can not send command sacct -n -j 11679249 -o "State" to mn1.bsc.es: SSH session not active
```

The point is that the autosubmit searcher web did not realized of that problem, and It was like this:

![Screenshot_colas_mn4](https://earth.bsc.es/gitlab/es/autosubmit/uploads/5a34e0b2f61fc12798ef10e251cf1508/Screenshot_colas_mn4.png)

But the user portal says:

![Screenshot_hpc_user_portal](https://earth.bsc.es/gitlab/es/autosubmit/uploads/e096bca61fb5fcdf4310a104639f0789/Screenshot_hpc_user_portal.png)

Actually, I've checked the output folder and it seems the job was completed. However, for a full day I thought I was in the queue because I only checked the autosubmit searcher web.

Why autosubmit searcher did not get updated? Why I had this error on the nohup?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2020, 09:15

Hi @mtplay, 

We're not certain of what conditions can cause this happen. We know that sometimes the ssh connection stops to work without Autosubmit being fully aware of the connection being stopped. We can mitigate this outcome to try to reconnect to the platforms once again(if there are not issues from remote platforms) but only on 3.13.0b.

In summary: 

In 3.13.0b this kind of issues are more detected and hopefully, it will be resolved in an automatic way.

In 3.12.0 this issue can't be resolved since the changes would be too large to make sense ( it would look similar to 3.13.0b) 

The solution on 3.12.0:

Stop autosubmit and run again the experiment. Autosubmit will sync with the remote platform job status. 

> Why autosubmit searcher did not get updated? Why I had this error on the nohup?

I guess that is due autosubmit being running all this time although it is spamming the session not active message. If autosubmit does not stop either will the GUI and GUI relies on autosubmit to update the status.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 2, 2020, 16:41

mentioned in issue auto-ecearth3#1413

**Comment by kinow** [comment]
In GitLab by @Eleftheriaexarchou on Oct 19, 2020, 12:57

I get this error frequently, for example now it happened for a33e, and this is why I had to use recovery which led to other issues  (see https://earth.bsc.es/gitlab/es/autosubmit/-/issues/600 as an example) can we prevent this from happening?

**Comment by kinow** [comment]
In GitLab by @mcastril on Nov 2, 2020, 13:33

closed
